lib/puppet_x/puppetlabs/property/region.rb

```diff
@@ -0,0 +1,15 @@
+module PuppetX
+  module Property
+    class AwsRegion < Puppet::Property
+      validate do |value|
+        name = resource[:name]
+        fail 'region should not contain spaces' if value =~ /\s/
+        fail 'region should not be blank' if value == ''
+        fail 'region should be a String' unless value.is_a?(String)
+        if !ENV['AWS_REGION'].nil? && ENV['AWS_REGION'] != value
```

Comment:
The logic is hard for me to read.  Is there another way that this can be written to provide a bit more clarity about the intent?